### ISSUE TITLE
[RFC 0069] Support Stack Buildpacks

### ISSUE THREAD

**Issue by ekcasey** [buildpacks/lifecycle#501]
[RFC 0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md)
https://github.com/buildpacks/rfcs/pull/111

https://github.com/buildpacks/spec/issues/177
https://github.com/buildpacks/spec/issues/178

**Comment by natalieparellano** [comment]
Blocked on https://github.com/buildpacks/lifecycle/issues/530 and (I think?) https://github.com/buildpacks/lifecycle/issues/471

This issue could probably benefit from being broken down into smaller issues, and having an associated spec PR at least in draft status (maybe the spec PR could also be broken down). 

Some thoughts on how this could be chunked based on the existing draft PR to lifecycle (https://github.com/buildpacks/lifecycle/pull/410):
* Static satisfaction (via `stack.toml`) of statically required (via `buildpack.toml`) mixins <- `pack` does this today, need to add to lifecycle
* Dynamic satisfaction (via build plan) of dynamically required (via build plan) mixins <- existing draft PR for stack buildpacks does this
* Dynamic satisfaction (via build plan) of statically required (via `buildpack.toml`) mixins <- seems this should be easier to add if we've done the first two
* Static satisfaction (via stack buildpacks' `buildpack.toml`) of statically required (via `buildpack.toml`) mixins
* Static satisfaction (via stack OR stack buildpacks' `buildpack.toml`) of dynamically required (via build plan) mixins
* A way for stackpacks to disable cache on rebuilds
* A way for stackpacks to skip taking new snapshots on subsequent re-runs
* A way for stackpacks to exclude/include parts of the snapshot
* `extend` phase
* Probably other things missing here

@jabrown85 what do you think?

**Comment by natalieparellano** [comment]
Also might be good to think about which of these issues could be parallelizable.

**Comment by natalieparellano** [comment]
Closing in favor of https://github.com/buildpacks/lifecycle/issues?q=is%3Aissue+label%3Aepic%2Fstackpacks+
